README.md

```diff
@@ -1,2 +1,422 @@
+
+Let‘s walk through each of these macros in detail.
+
+### @Instantiable
+
+Type declarations decorated with the [`@Instantiable` macro](Sources/SafeDI/PropertyDecoration/Instantiable.swift) are able to be instantiated by SafeDI. Types decorated with this macro can instantiate other `@Instantiable` dependencies, forward dependencies injected from outside of SafeDI, or receive dependencies instantiated or forwarded by objects further up the dependency tree.
+
+Every `@Instantiable`-decorated type must be:
+
+1. `public` or `open`
+
+2. Have a `public init(…)` or `open init(…)` method that receives every injectable property
+
+The `@Instantiable` guides engineers through satisfying these requirements with build-time FixIts.
+
+#### Example
+
+Here is a sample `UserService` implementation that is `@Instantiable`:
+
+```swift
+import SafeDI
+
+/// A protocol that defines a UserService.
+/// It is not necessary to utilize protocols with SafeDI, but since
+/// protocol-driven development aids both testability and dependency
+/// inversion, our examples show how protocols can be used with SafeDI.
+public protocol UserService {
+    var user: User? { get }
+    func login(username: String, password: String) async throws -> User
+}
+
+/// A default implementation of `UserService` that can fulfill `@Instantiated`
+/// properties of type `UserService` or `DefaultUserService`.
+@Instantiable(fulfillingAdditionalTypes: [UserService.self])
+public final class DefaultUserService: UserService {
+
+    // MARK: Initialization
+
+    /// Public, memberwise initializer that takes each injected property.
+    public init(authService: AuthService, securePersistentStorage: SecurePersistentStorage) {
+        self.authService = authService
+        self.securePersistentStorage = securePersistentStorage
+    }
+
+    // MARK: UserService
+
+    public private(set) lazy var user: User? = loadPersistedUser() {
+        didSet {
+            persistUserData()
+        }
+    }
+
+    public func login(username: String, password: String) async throws -> User {
+        let user = try await authService.login(username: username, password: password)
+        self.user = user
+        return user
+    }
+
+    // MARK: Private
+
+    /// An auth service instance that is instantiated when the `DefaultUserService` is instantiated.
+    @Instantiated
+    private let authService: AuthService
+
+    /// An instance of secure, persistent storage that is instantiated further up the SafeDI dependency tree.
+    @Received
+    private let securePersistentStorage: SecurePersistentStorage
+
+    private func loadPersistedUser() -> User? {
+        securePersistentStorage["user", ofType: User.self]
+    }
+
+    private func persistUserData() {
+        securePersistentStorage["user"] = user
+    }
+}
+```
+
+### @ExternalInstantiable
```

Comment:
I think you should keep the same order as in the intro, so `@ExternalInstantiable` should come after `@Received`

Reply:
Per Kieraj above I'm likely to collapse `@ExternalInstantiable` into `@Instantiable` which will simplify the ordering a bit.